firmware/src/opaygo_functions.hpp

```diff
@@ -0,0 +1,313 @@
+#pragma once
+
+// OpenSmartMeter libraries
+#include "global_defines.hpp"
+#include "helpers.hpp"
+#include "mem_init.hpp"
+#include "time_management.hpp"
+
+// Arduino base libraries
+#include <inttypes.h>
+#include "Arduino.h"
+// #include <time.h>
+
+extern "C" {
+#include "opaygo_decoder.h"
+}
+
+#define BLINK_PERIOD 250
+#define DEBUG
+
+#define STAR_KEY -1
+#define HASH_KEY -2
+#define NON_ACCEPTED_KEY -3
+
+#define TIME_DIVIDER 1
+// #define RESTRICTED_DIGIT_SET_MODE
+
+#ifdef RESTRICTED_DIGIT_SET_MODE
+#  define TOKEN_LENGTH 15
+#else
+#  define TOKEN_LENGTH 9
+#endif
+
+// Device parameters location in Flash/EEPROM
+unsigned int TokenCount_eeprom_location = 6;
+unsigned int UsedTokens_eeprom_location = 8;
+unsigned int PAYGEnabled_eeprom_location = 10;
+unsigned int ActiveUntil_eeprom_location = 11;
+unsigned int TokenEntryLockedUntil_eeprom_location = 15;
+unsigned int LAST_TIME_STAMP_ADDRESS = 27;
+unsigned int NB_DISCONNECTIONS_ADDRESS = 32;
```

Comment:
What do you think about making some like a "global" EEPROM map? We don't have to do this now in this PR, but as an idea?

For example, in `global_defined` we could include all the EEPROM locations as macros?